### ISSUE TITLE
Investigate shm indel in QA255.157-Vk

### ISSUE THREAD

**Issue by psathyrella** [matsengrp/cft#279]
original partis full run had 2800 sequences in the family, laura manually pulled out the ones with 3base shm deletion, got 317. Where are these sequences/why aren't they all in the output of the indeld/non_indeld study.

**Comment by psathyrella** [comment]
So as I was suggesting this is because there's a very large number of sequences in a lineage that is close to, and maybe part of, the seed lineage. The easiest way to see is just annotate the 317 sequences that laura pulled out, partial screenshot here:

![p](https://user-images.githubusercontent.com/2960243/58673223-553f6480-82ff-11e9-83c3-947bb9fb218a.png)

full output attached [here](https://github.com/matsengrp/cft/files/3239184/laura-317-seqs.log). What I'm looking at is that there's little shared mutation, and quite different cdr3, between the seed (.157) and the majority of the sequences.

To see in more detail, the ascii from [v18](https://github.com/matsengrp/cft/files/3239187/v18.log) (with several thousand) and [v19](https://github.com/matsengrp/cft/files/3239188/v19.log) (more similar to the version that we were doing the indel study with).

tl;dr is that light chain families basically always consist of many different widely separated branches/sublineages, that maybe or maybe not connect at the tippy top near the root. Or, since I like drawing pictures:

![IMG_20190530_172813](https://user-images.githubusercontent.com/2960243/58673537-9dab5200-8300-11e9-8e0c-0b37752e79f2.jpg)

**Comment by metasoarous** [comment]
@psathyrella @lauradoepker Sounds then like this is a somewhat unavoidable result of the data. Is some further action on this warranted, or shall we close this?

**Comment by matsen** [comment]
Thanks for the screenshot, Duncan. I agree with you that this isn't an easy problem. Though if I'm reading the diagram correctly it looks to me that these sequences do share a bunch of mutation signal:

![sub-screenshot](https://user-images.githubusercontent.com/112708/58703305-150ecf00-835d-11e9-83e5-fcd36565c304.png)

It also looks to me that `QA255-k-d1174-j18-40943-1` is relatively unmutated compared to the rest of these sequences. Could that lead us to the root sequence?

**Comment by psathyrella** [comment]
Right, there are sequences that share lots of mutations with the seed -- those are the smaller subcluster. But that smaller subcluster doesn't share a lot of mutations with the rest of that ^ large cluster. I maybe should have made the screenshot bigger, so you could see that the sequences similar to the seed (in the upper third/half of my screenshot) are like 10% of the whole annotation (in the attached file).

Yes, that's a good point. The existence of a sequence with only like 3 mutations means that linearham should be smart enough to pick up on the fact that those bands of shared mutation on the right side of the cdr3 are probably actually in the naive cdr3. Although in the 10% subcluster partis already picks up on much of that, so it might not be a big difference:

![p](https://user-images.githubusercontent.com/2960243/58723550-3092cd80-838f-11e9-8cc3-8d8c1a601f6c.png)

And I guess more generally you can see from this that we'll probably do pretty well with the naive inference, since it's still pretty big (325 sequences) and there's lots of diverse mutation. Although that's all with the indel in -- if we restrict to sequences that align, without reverting the indel, to the other V gene, then we have way fewer.

**Comment by matsen** [comment]
The plan, after meeting:

* add an option to process_partis that tells it to pull out all the sequences in a given cluster that have "compatible" indels to a given sequence (#281)
* then let's build an ML tree for those pulled out sequences, in their non-indel-reversed form (will require MSA). We'll need to do this with pruning, etc.
* also do ML phylogenetics on the forbidding-indels cluster

Look at sequences close to the root in each case with Olmsted. We'll have to make a judgement call.

**Comment by psathyrella** [comment]
no-indel full rerun version:  `/fh/fast/matsen_e/processed-data/partis/qa255-synth/no-indels/info.yaml`
[no-indels.log](https://github.com/matsengrp/cft/files/3242672/no-indels.log)

**Comment by lauradoepker** [comment]
@psathyrella @matsen doesn't the fact that these sequences all share the exact same deletion event speak in their favor of being clonal, rather than two separate families that managed the same unlikely deletion?  We're getting into statistical probability here... and we likely don't know well enough to say what's likely and unlikely to repeat in indel world.

**Comment by matsen** [comment]
Yes, all the hypotheses we are considering have the same deletion event in a single clonal family. The question is if we align to:

* a germline gene that has that deletion but requires more point mutation
* a germline gene that doesn't have that deletion, but requires less point mutation (but yes, a deletion SHM)

Correct me, @psathyrella , if I'm getting it wrong.

**Comment by eharkins** [comment]
@lauradoepker Ok to close?

**Comment by lauradoepker** [comment]
Yep!
